File: frontend/src/app/controller/service/admin/gestionchequeffet/ChequeEffetAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {ChequeEffetDto} from 'src/app/controller/model/gestionchequeffet/ChequeEffet.model';
import {ChequeEffetCriteria} from 'src/app/controller/criteria/gestionchequeffet/ChequeEffetCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ChequeEffetAdminService extends AbstractService<ChequeEffetDto, ChequeEffetCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/chequeEffet/');
    }

    public constrcutDto(): ChequeEffetDto {
        return new ChequeEffetDto();
    }

    public constrcutCriteria(): ChequeEffetCriteria {
        return new ChequeEffetCriteria();
    }
}

File: backend-ms1/src/main/java/ma/zs/easystock/service/facade/admin/gestionchequeffet/ChequeEffetAdminService.java
package ma.zs.easystock.service.facade.admin.gestionchequeffet;

import java.util.List;
import ma.zs.easystock.bean.core.gestionchequeffet.ChequeEffet;
import ma.zs.easystock.dao.criteria.core.gestionchequeffet.ChequeEffetCriteria;
import ma.zs.easystock.zynerator.service.IService;



public interface ChequeEffetAdminService extends  IService<ChequeEffet,ChequeEffetCriteria>  {

    List<ChequeEffet> findByFournisseurId(Long id);
    int deleteByFournisseurId(Long id);
    long countByFournisseurTelephone(String telephone);



}
